### ISSUE TITLE
"conda build" is not found even if conda-build is installed. (miniconda update issue?)

### ISSUE THREAD

**Issue by Callek** [conda/conda#12955]
### What happened?

```
$ conda build --help
usage: conda [-h] [--no-plugins] [-V] COMMAND ...
conda: error: argument COMMAND: invalid choice: 'build' (choose from 'clean', 'compare', 'config', 'create', 'info', 'init', 'install', 'list', 'notices', 'package', 'remove', 'uninstall', 'rename', 'run', 'search', 'update', 'upgrade', 'doctor', 'content-trust', 'env')
```

(note `conda-build --help` works fine)

### Conda Details

<details>
<summary><code>conda info</code></summary>

```shell
     active environment : sdk-build-env
    active env location : /tmp/asd/envs/sdk-build-env
            shell level : 1
       user config file : /Users/callek/.condarc
 populated config files : /Users/callek/.condarc
          conda version : 23.7.2
    conda-build version : not installed
         python version : 3.9.17.final.0
       virtual packages : __archspec=1=arm64
                          __osx=13.4.1=0
                          __unix=0=0
       base environment : /tmp/asd  (writable)
      conda av data dir : /tmp/asd/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /tmp/asd/pkgs
                          /Users/callek/.conda/pkgs
       envs directories : /tmp/asd/envs
                          /Users/callek/.conda/envs
               platform : osx-arm64
             user-agent : conda/23.7.2 requests/2.31.0 CPython/3.9.17 Darwin/22.5.0 OSX/13.4.1
                UID:GID : 502:20
             netrc file : None
           offline mode : False
```

</details>

<details>
<summary><code>conda config</code></summary>

```shell
==> /Users/callek/.condarc <==
always_yes: True

==> cmd_line <==
debug: False
json: False
```

</details>

<details>
<summary><code>conda list</code></summary>

```shell
# packages in environment at /tmp/asd/envs/sdk-build-env:
#
# Name                    Version                   Build  Channel
beautifulsoup4            4.12.2          py311hca03da5_0    defaults
boltons                   23.0.0          py311hca03da5_0    defaults
brotlipy                  0.7.0           py311h80987f9_1002    defaults
bzip2                     1.0.8                h620ffc9_4    defaults
ca-certificates           2023.05.30           hca03da5_0    defaults
cctools                   949.0.1             hc179dcd_25    defaults
cctools_osx-arm64         949.0.1             h332cad3_25    defaults
certifi                   2023.7.22       py311hca03da5_0    defaults
cffi                      1.15.1          py311h80987f9_3    defaults
chardet                   4.0.0           py311hca03da5_1003    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
click                     8.0.4           py311hca03da5_0    defaults
conda                     23.7.2          py311hca03da5_0    defaults
conda-build               3.25.0          py311hca03da5_0    defaults
conda-index               0.2.3           py311hca03da5_0    defaults
conda-package-handling    2.1.0           py311hca03da5_0    defaults
conda-package-streaming   0.8.0           py311hca03da5_0    defaults
cryptography              41.0.2          py311h6204c90_0    defaults
filelock                  3.9.0           py311hca03da5_0    defaults
glob2                     0.7                pyhd3eb1b0_0    defaults
icu                       68.1                 hc377ac9_0    defaults
idna                      3.4             py311hca03da5_0    defaults
jinja2                    3.1.2           py311hca03da5_0    defaults
jsonpatch                 1.32               pyhd3eb1b0_0    defaults
jsonpointer               2.1                pyhd3eb1b0_0    defaults
ld64                      530                 hb29bf3f_25    defaults
ld64_osx-arm64            530                 h001ce53_25    defaults
ldid                      2.1.5                h20b2a84_3    defaults
libarchive                3.6.2                h62fee54_2    defaults
libcxx                    14.0.6               h848a8c0_0    defaults
libffi                    3.4.4                hca03da5_0    defaults
libiconv                  1.16                 h1a28f6b_2    defaults
liblief                   0.12.3               h313beb8_0    defaults
libllvm14                 14.0.6               h7ec7a93_3    defaults
libxml2                   2.10.3               h372ba2a_0    defaults
lz4-c                     1.9.4                h313beb8_0    defaults
markupsafe                2.1.1           py311h80987f9_0    defaults
more-itertools            8.12.0             pyhd3eb1b0_0    defaults
ncurses                   6.4                  h313beb8_0    defaults
openssl                   3.0.9                h1a28f6b_0    defaults
packaging                 23.0            py311hca03da5_0    defaults
patch                     2.7.6             h1a28f6b_1001    defaults
pkginfo                   1.9.6           py311hca03da5_0    defaults
pluggy                    1.0.0           py311hca03da5_1    defaults
psutil                    5.9.0           py311h80987f9_0    defaults
py-lief                   0.12.3          py311h313beb8_0    defaults
pycosat                   0.6.4           py311h80987f9_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyopenssl                 23.2.0          py311hca03da5_0    defaults
pysocks                   1.7.1           py311hca03da5_0    defaults
python                    3.11.4               hb885b13_0    defaults
python-libarchive-c       2.9                pyhd3eb1b0_1    defaults
pytz                      2022.7          py311hca03da5_0    defaults
pyyaml                    6.0             py311h80987f9_1    defaults
readline                  8.2                  h1a28f6b_0    defaults
requests                  2.31.0          py311hca03da5_0    defaults
ruamel.yaml               0.17.21         py311h80987f9_0    defaults
setuptools                68.0.0          py311hca03da5_0    defaults
six                       1.16.0             pyhd3eb1b0_1    defaults
soupsieve                 2.4             py311hca03da5_0    defaults
sqlite                    3.41.2               h80987f9_0    defaults
tapi                      1100.0.11            h8754e6a_1    defaults
tk                        8.6.12               hb8d0fd4_0    defaults
toolz                     0.12.0          py311hca03da5_0    defaults
tqdm                      4.65.0          py311hb6e6a13_0    defaults
tzdata                    2023c                h04d1e81_0    defaults
urllib3                   1.26.16         py311hca03da5_0    defaults
xz                        5.4.2                h80987f9_0    defaults
yaml                      0.2.5                h1a28f6b_0    defaults
zlib                      1.2.13               h5a0b063_0    defaults
zstandard                 0.19.0          py311h80987f9_0    defaults
zstd                      1.5.5                hd90d995_0    defaults
```

</details>

### Additional Context

The above was with an explicit `conda-build=3.25.0` though it reproduces with latest

### Steps to Reproduce ###

Requires two terminals, neither of which have conda activated in the shell (achieved by commenting out all conda stuff in `~/.bash_profile`) and reproduced on an arm OSX system (but first experienced it in automation on linux-x86).

#### Terminal 1 ####

```
 # also reproduced with 23.1.0-1
wget https://repo.anaconda.com/miniconda/Miniconda3-py39_23.5.2-0-MacOSX-arm64.sh
sh ./Miniconda3-py39_23.5.2-0-MacOSX-arm64.sh -b -p /tmp/asd

source "/tmp/asd/etc/profile.d/conda.sh"
conda install conda-libmamba-solver -y --solver=classic
conda update conda -y
conda activate base
conda config --set always_yes yes
```

#### Terminal 2 ####

```
# After all steps from terminal 1

source /tmp/asd/etc/profile.d/conda.sh
unset CONDA_EXPERIMENTAL_SOLVER
mkdir reproducer
# Also reproduces with conda-build=23.3.0
conda create --name reproducer-env -y conda-build
conda activate reproducer-env
conda build --help
```

**Comment by maximlt** [comment]
Observed the same symptoms on our CI two days ago https://github.com/holoviz/param/actions/runs/5706463077/job/15462166474

**Comment by kenodegard** [comment]
Ugh I have a suspicion for why this is

In the meantime to fix this either continue using `conda-build` or update to `conda-build 3.26.0`

**Comment by Callek** [comment]
> Ugh I have a suspicion for why this is
> 
> In the meantime to fix this either continue using `conda-build` or update to `conda-build 3.26.0`

Same symptoms reproduced on `conda-build` 3.26.0 as well.

**Comment by picoutputcls** [comment]
I've run into a similar issue to this with the `conda render` command in respect to `conda-build`. Running a `conda render ..` command with `conda` v`23.7.2` returns an error like:
`conda: error: argument COMMAND: invalid choice: 'render' (choose from 'clean', 'compare', 'config', 'create', 'info', 'init', 'install', 'list', 'notices', 'package', 'remove', 'uninstall', 'rename', 'run', 'search', 'update', 'upgrade', 'doctor', 'content-trust', 'tree', 'env', 'lock')`

I imagine the behaviour is the same with any other `conda-build` related commands. 

Running `conda-render ...` works as per the similar behaviour described above for `conda build`. But for now I've downgraded `conda` to v`23.5.2` and everything seems to be working as expected again.

**Comment by jezdez** [comment]
Could you try reproducing this by installing conda-build in your conda base environment?

**Comment by Callek** [comment]
> Could you try reproducing this by installing conda-build in your conda base environment?

Using my original steps to reproduce, added `conda install conda-build` to the end of the Terminal 1 steps.

Once complete, the steps in Terminal 2 resulted in the `conda build` help message as expected, rather than this error condition.

**Comment by kenodegard** [comment]
Resolved in #13006 and will be fixed in `conda 23.7.3`
